CompositionRoot/B.swift

```diff
@@ -1,17 +1,17 @@
-protocol BPrototol {
+protocol BProtocol {
     func bMethod()
 }
 
-class B: BPrototol {
+class B: BProtocol {
     
-    private let applicationContext: ApplicationContext
+    private let cFactory: CFactoryProtocol
     
-    init(applicationContext: ApplicationContext) {
-        self.applicationContext = applicationContext
+    init(cFactory: CFactoryProtocol) {
+        self.cFactory = cFactory
     }
     
     func bMethod() {
-        let c = C(applicationContext: applicationContext)
+        let c = cFactory.createC()
```

Comment:
B now depends on interfaces completely.  cFactory is of type `CFactoryProtocol`.

Reply:
Inject cFactory make it easy to replace CFactory with FakeCFactory, so it's easy to unit test this function.